src/Symfony/Component/Console/Command/Command.php

```diff
@@ -554,11 +572,14 @@ public function getProcessedHelp()
      */
     public function setAliases(iterable $aliases)
     {
+        $list = [];
+
         foreach ($aliases as $alias) {
             $this->validateName($alias);
+            $list[] = $alias;
         }
 
-        $this->aliases = $aliases;
+        $this->aliases = \is_array($aliases) ? $aliases : $list;
```

Comment:
Question: Do we need the is_array?  We are now building $list and it is going to be an array.  I think the check is ensuring $this->aliases is set to an array.  As $list is always an array and it is guaranteed to contain the same as $aliases can we just use that all the time?
```
 $this->aliases = $list;
```

Reply:
This is an optimization trick that saves memory: when then variable is an array, $list is a hard copy of $aliases, while $aliases is a copy-on-write reference.